### PR TITLE
Assign container name based on image name for oc create dc

### PR DESCRIPTION
**Pull request by nak3**
Currently `oc create dc` always assign container name as
`default-container`.
It could be useful to have more recognizable name by default.

This patch changes to assign the container name based on
the specified image name.

Before:
```
$ oc create dc nginx --image=nginx -o yaml --dry-run
  ...
      containers:
      - image: nginx
        name: default-container
        resources: {}
```

After:
```
$ oc create dc nginx --image=nginx -o yaml --dry-run
  ...
      containers:
      - image: nginx
        name: nginx
        resources: {}
```

### CODE REVIEW

File: pkg/oc/cli/cmd/create/deploymentconfig.go [hunk 1]
```diff
@@ -122,6 +125,21 @@ func (o *CreateDeploymentConfigOptions) Complete(cmd *cobra.Command, f *clientcm
 	return nil
 }
 
+// parseImageName parses the image strings and returns just the image name
+func parseImageName(imageString string) string {
+	// Retain just the image name
+	imageSplit := strings.Split(imageString, "/")
+	name := imageSplit[len(imageSplit)-1]
+	// Remove any tag or hash
+	if strings.Contains(name, ":") {
```
Comment (nak3) [inline pkg/oc/cli/cmd/create/deploymentconfig.go]: This code is based on `kubectl create deployment`. However, upstream seems to have an issue https://github.com/kubernetes/kubernetes/issues/62252 so here has a minor update which removes `else if`.

**Comment by nak3** [comment]
> It could be useful to have more recognizable name by default.

When multiple images will be added, the container name should be recognizable. Also, I faced this trouble when set imageChange triggers.

1. Create a DC

~~~
$ oc create dc nginx --image=nginx
deploymentconfig "nginx" created
~~~

2. Try to set a trigger with imageChange (`default-container` have to be assigned)

~~~
$ oc set triggers dc nginx --from-image=nginx:latest
error: deploymentconfigs/nginx you must specify --containers when setting --from-image
~~~

`--containers` needs to be specified, so I found `default-container`. (But it was so strange name and I had to start reserching it...)

File: pkg/oc/cli/cmd/create/deploymentconfig.go (hunk 1 above)
Comment (smarterclayton) [reply pkg/oc/cli/cmd/create/deploymentconfig.go]: Please don't parse the image name like this.  We should be using the real name parsing logic that we have elsewhere.

/hold

**Comment by smarterclayton** [comment]
Removing label, we shouldn't be writing custom image parsing logic.  Also, not all image names are valid deployment config names, so your logic will fail in some cases.

**Comment by nak3** [comment]
Thank you. Updated.

> Also, not all image names are valid deployment config names, so your logic will fail in some cases.

This PR is getting `container name` from image name. I think that all image names are valid for container name. (If not, I guess that oc new-app have to be updated as well.)

**Pull request closed without merging**